test/service/test_portal_service.py

```diff
@@ -143,7 +131,11 @@ def test_internal_crud(self):
         with self.subTest('read'):
             read_db = self.portal_service._read_db(version)
             self.assertEqual(read_db, download_db)
-            self.assertRaises(NoSuchObjectVersion, self.portal_service._read_db, 'fake_version')
+            # The versionID below is encoded correctly, but does not belong to
+            # any s3 object. https://github.com/spulec/moto/issues/3884
+            self.assertRaises(NoSuchObjectVersion,
+                              self.portal_service._read_db,
+                              'VWVT9JkWTreQ95JbRmQt6T3LWrljLpRZ')
```

Comment:
Q for @hannes-ucsc

Not sure if this is a required change, however it would more close to an exception from the s3api.

Reply:
update to `3sL4kqtJlcpXroDTDmJ+rmSpXd3dIbrHY+MTRCxf3vjVBH40Nr8X8gdRQBpUMLUo`

add link to https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html